tests/bsim/bluetooth/audio/src/has_client_test.c

```diff
@@ -229,25 +246,448 @@ static void test_main(void)
+		subscribe_params->notify = notify_handler;
+		subscribe_params->value = BT_GATT_CCC_INDICATE;
+		subscribe_params->ccc_handle = attr->handle;
+		subscribe_params->subscribe = subscribe_cb;
+
+		err = bt_gatt_subscribe(conn, subscribe_params);
+		if (err && err != -EALREADY) {
+			printk("Subscribe failed (err %d)\n", err);
+		}
+	} else {
+		printk("Unknown handle at %d\n", attr->handle);
+		return BT_GATT_ITER_CONTINUE;
+	}
+
+	return BT_GATT_ITER_STOP;
+}
+
+static void discover_and_subscribe_control_point(void)
+{
+	int err = 0;
+
+	printk("%s\n", __func__);
+
+	memcpy(&uuid, BT_UUID_HAS_PRESET_CONTROL_POINT, sizeof(uuid));
+	discover_params.uuid = &uuid.uuid;
+	discover_params.start_handle = BT_ATT_FIRST_ATTRIBUTE_HANDLE;
+	discover_params.end_handle = BT_ATT_LAST_ATTRIBUTE_HANDLE;
+	discover_params.type = BT_GATT_DISCOVER_CHARACTERISTIC;
+	discover_params.func = discover_control_point_cb;
+
+	err = bt_gatt_discover(default_conn, &discover_params);
+	if (err != 0) {
+		FAIL("Control Point failed (err %d)\n", err);
+		return;
+	}
+}
+
+static void test_gatt_client(void)
+{
+	int err;
+
+	err = bt_enable(NULL);
+	if (err < 0) {
+		FAIL("Bluetooth discover failed (err %d)\n", err);
+		return;
+	}
+
+	printk("Bluetooth initialized\n");
+
+	bt_le_scan_cb_register(&common_scan_cb);
+
+	err = bt_le_scan_start(BT_LE_SCAN_PASSIVE, NULL);
+	if (err < 0) {
+		FAIL("Scanning failed to start (err %d)\n", err);
+		return;
+	}
+
+	printk("Scanning successfully started\n");
+
+	WAIT_FOR_FLAG(flag_connected);
+
+	err = bt_conn_set_security(default_conn, BT_SECURITY_L2);
+	if (err) {
+		FAIL("Failed to set security level %d (err %d)\n", BT_SECURITY_L2, err);
+		return;
+	}
+
+	WAIT_FOR_COND(security_level == BT_SECURITY_L2);
+
+	discover_and_subscribe_features();
+	WAIT_FOR_FLAG(flag_features_discovered);
+
+	discover_and_subscribe_active_preset_index();
+	WAIT_FOR_FLAG(flag_active_preset_index_discovered);
+
+	discover_and_subscribe_control_point();
+	WAIT_FOR_FLAG(flag_control_point_discovered);
+
+	bt_conn_disconnect(default_conn, BT_HCI_ERR_REMOTE_USER_TERM_CONN);
+	WAIT_FOR_UNSET_FLAG(flag_connected);
+
+	notify_received_mask = 0;
+	UNSET_FLAG(flag_all_notifications_received);
+
+	err = bt_le_scan_start(BT_LE_SCAN_PASSIVE, NULL);
+	if (err < 0) {
+		FAIL("Scanning failed to start (err %d)\n", err);
+		return;
+	}
+
+	printk("Scanning successfully started\n");
+
+	WAIT_FOR_FLAG(flag_connected);
+
+	err = bt_conn_set_security(default_conn, BT_SECURITY_L2);
+	if (err) {
+		FAIL("Failed to set security level %d (err %d)\n", BT_SECURITY_L2, err);
+		return;
+	}
+
+	WAIT_FOR_FLAG(flag_all_notifications_received);
+
+	PASS("HAS main PASS\n");
+}
 
 static const struct bst_test_instance test_has[] = {
+#ifdef CONFIG_BT_HAS_CLIENT
 	{
 		.test_id = "has_client",
 		.test_post_init_f = test_init,
 		.test_tick_f = test_tick,
 		.test_main_f = test_main,
 	},
+#endif /* CONFIG_BT_HAS_CLIENT */
+	{
+		.test_id = "has_client_gatt",
```

Comment:
The `_gatt` suffix isn't very descriptive. If you want to keep the name, please consider adding some text to `.test_descr`